src/MKL.jl

```diff
@@ -1,9 +1,53 @@
 module MKL
 
-using MKL_jll
+using Preferences
+using Libdl
 
-JULIA_VER_NEEDED = v"1.7.0-DEV.641"
-VERSION > JULIA_VER_NEEDED && using LinearAlgebra
+# Choose an MKL provider; taking an explicit preference as the first choice,
+# but if nothing is set as a preference, fall back to an environment variable,
+# and if that is not given, fall back to the default choice of `MKL_jll`.
+const mkl_provider = lowercase(something(
+    @load_preference("mkl_provider", nothing),
+    get(ENV, "JULIA_MKL_PROVIDER", nothing),
+    "mkl_jll",
+)::String)
+
+if mkl_provider == "mkl_jll"
+    # Only load MKL_jll if we are suppoed to use it as the MKL source
+    # to avoid an unnecessary download of the (lazy) artifact.
+    import MKL_jll
+    const libmkl_rt = MKL_jll.libmkl_rt
+    const mkl_path = dirname(libmkl_rt)
+elseif mkl_provider == "system"
```

Comment:
Could we combine `mkl_provider` and `mkl_path` such that instead of setting `mkl_provider = "system"`, we instead set `mkl_provider = "/path/to/libmkl_rt.so"`?
The "special" value of `mkl_provider == "mkl_jll"` could stay, but we could just assume that anything else is a path and try to open it.

Reply:
I still kept `mkl_path == "system"` as a special case for the "try to automatically find the libmkl_rt" option, which may still be useful. Everything else is assumed to be a valid path to libmkl_rt from now on.